[wj331] iP

wj331 opened the pull request:
# TaskList Chatbot <Dick>
```
               " ____    __        __       \n"
            + "|  _ \\   | |  ____ | | __\n"
            + "| | | |  | | |     | |/ /\n"
            + "| |_| |  | | |     |   < \n"
            + "|____/   ___  ____ |_|\\_\\\n";
```
> The one stop checklist bot to track all your tasks.

**USE** this bot to keep track of your **DEADLINES**, **TODOS** and **EVENTS**

##With this bot, you can do the following : 
1. Add Tasks like ToDos, Deadlines and Events into your List
2. Remove Tasks like lectures which u ~~suddenly~~ don't need to attend
3. Mark / Unmark Tasks which you have done & undone
4. Find Task which based on your **KEYWORD** input
5. **AND MORE!!**

To Set Up this bot, refer to this [link](https://nus-cs2103-ay2324s1.github.io/website/schedule/week2/project.html)

## Example ways to use the bot :
### - Creating a **ToDo** task
Type `todo` followed by the name of your todo task
For example:
```todo read book```

### - Creating a **Deadline** task
Type `deadline` followed by the name and `/by _deadline_`
For example
```deadline return book /by Sunday```

### - Creating an **Event** task
Type `event` followed by the name, and `/from startTime /to endTime`
For example
```event kpop /from 12pm /to 2pm```


With that, I hope you enjoy using this bot! 💯

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

wj331 committed:
Level-0

wj331 committed:
Level-1

wj331 committed:
Level-2

wj331 committed:
Level-3

wj331 committed:
Level-4

wj331 committed:
TextUiTesting

wj331 committed:
Level-5

wj331 committed:
Level-6

wj331 committed:
Updated ui testing

wj331 committed:
Level 7 completed

wj331 committed:
Merge branch 'branch-Level-7'

wj331 committed:
Level 8 completed

wj331 committed:
A-MoreOOP done

wj331 committed:
A-Package done

wj331 committed:
Merge branch 'A-Gradle'

wj331 committed:
A-Gradle done

wj331 committed:
Merge branch 'branch-A-Gradle'

wj331 committed:
A-JUnit

wj331 committed:
Merge branch 'branch-A-JUnit'

wj331 committed:
A-JavaDoc done

wj331 committed:
branch A-Jar completed

wj331 committed:
A-CodingStandard done

wj331 committed:
Level-9 done + bug fixes

bwangpj commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,114 @@
+    }
+
+    /**
+     * To write tasks from TaskList into the file
```
Comment should start with "Writes" instead of "To write"?

bwangpj commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,114 @@
+        }
+    }
+    /**
+     * To load an existing file into a new TaskList object.
```
Comment should start with "Loads" instead of "To load"?

bwangpj commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,114 @@
+     * @throws InvalidInputException an exception thrown whenever input is invalid
+     */
+    public List<Task> load() throws InvalidInputException {
+        List<Task> lst = new ArrayList<>();
```
Perhaps a more descriptive variable name than 'lst'?

bwangpj commented on src/main/java/duke/ToDo.java:
```diff
@@ -0,0 +1,35 @@
+ * To-Do class that has 2 fields, its symbol and name
+ */
+public class ToDo extends Task {
+    private String symbol = "[T]";
```
If this is a constant, variable name should be in uppercase?

bwangpj reviewed (commented):
Coding standard PR review as per Week 4 requirements.

javinchua commented on src/main/java/duke/DeleteTask.java:
```diff
@@ -0,0 +1,48 @@
+ * @author wj331
+ */
+public class DeleteTask extends Command {
+    private int indexToDelete;
```
```suggestion
    private int index;
```
Perhaps naming it as index could be more concise and still clear.

javinchua commented on src/main/java/duke/DoneTask.java:
```diff
@@ -0,0 +1,34 @@
+ * @author wj331
+ */
+public class DoneTask extends Command {
+    private int taskDone;
```
```suggestion
    private int taskIndex;
```
taskDone as a variable name might imply a boolean (task done or not). A clearer name could be taskIndexToMarkDone or simply taskIndex.

javinchua commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,78 @@
+ * Task class that contains field such as name, completed and done & undone checkboxes
+ */
+abstract class Task {
+    private static String doneCheckbox = "[X] ";
+    private static String undoneCheckbox = "[ ] ";
```
Consider making them as final since they are constants.

javinchua commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,78 @@
+    private static String doneCheckbox = "[X] ";
+    private static String undoneCheckbox = "[ ] ";
+    private String name;
+    private boolean completed;
```
```suggestion
    private boolean isCompleted;
```

javinchua reviewed (commented):
Overall, enjoy the abstraction of the duke class. Consider putting the classes into different packages based on what they do, for instance Deadline and Event can be in a tasks package while The AddTask or FindTask classes can be in a command package.

wj331 committed:
Level-10 done

wj331 committed:
Checkstyle & javadocs mods

wj331 committed:
Add Java Assert to DoneTask, DeleteTask, AddTask

wj331 committed:
Merge pull request #1 from wj331/branch-A-Assertions

Add Java Assert to DoneTask, DeleteTask, AddTask

wj331 committed:
Adjusted Code Quality of some classes

wj331 committed:
Merge pull request #2 from wj331/branch-A-CodeQuality

Adjusted Code Quality of some classes

wj331 committed:
Add a update feature to update tasks' name/deadline/timings

wj331 committed:
Merge pull request #3 from wj331/branch-C-Update

Branch C Update

wj331 committed:
Implement update function

Bot now allows updating of task with a update <index> <details>.
Also, throws error if insufficient arguments encountered, and cancels
the update operation.

Minor bug fixes regarding not trimming input of user previously

wj331 committed:
Merge pull request #4 from wj331/branch-A-Streams

Using Streams in FindTask

wj331 committed:
Implemented exception handling for FindTask

Such as when there are no matching Tasks found

wj331 committed:
update storage and bye command

wj331 committed:
update JUnit testing

wj331 committed:
update User Guide in README

wj331 committed:
fix checkstyle and javadocs

wj331 committed:
Update README.md

wj331 committed:
Update README.md

wj331 committed:
Update README.md

wj331 committed:
update readme

wj331 committed:
Merge branch 'master' of https://github.com/wj331/ip

wj331 committed:
change from java 17 to java 11

wj331 committed:
Update README.md
